### DISCUSSION TITLE
Using PedalinoMini to send notes with varying Velocity

### DISCUSSION THREAD

**Discussion by Steffinuss** [alf45tar/PedalinoMini#476] [Q&A]
I already use the PedalinoMini as a controller pedal for guitar effects and the LoopyHD App. Thank you very much for this incredible work!
Now my son started playing the drums and I would like to build a custom electronic drumset for him. I am wondering if the PedalinoMini could also be useful for this built. I already managed to create actions to trigger notes for base drum, snare and hi-hat in garageband (Press-Release Note ON/OFF) with my existing PedalinoMini pedalboard:

<img width="496" alt="Bildschirmfoto 2023-11-26 um 23 28 17" src="https://github.com/alf45tar/PedalinoMini/assets/30431113/8bb571ed-2700-4471-9819-6dd3e120eb34">

Of course the buttons I have now are not pressure sensitive, thus the velocity is set to a certain number specified in the PedalinoMini Web Interface, in this case 127.

My main question now is: Is there any way to use the Pedalino Mini to send "Note On" MIDI commands with varying velocities?
If possible I could use the PedalinoMini to create my electric drum kit based on piezo drum pads. However, I am not sure if there is any action in the web interface that outputs a "note on" command in combination with variable velocity depending on a certain analog input.

**Comment by alf45tar** [comment]
I suggest to test the following setup

- Connect the piezo to an Analog pedal (schematic to be determined but it looks like a MOhm resistor in parallel could be enough)
- Define an Analog pedal without Automatic Calibration and use Min as threshold for triggering
- Define an Action on pedal Move as NoteOn
- When piezo pressure is greater than Min value it is translated into note velocity

If you plan to do some testing let us know the results.

**Reply by Steffinuss** [reply]
Thank you very much for your suggestions! I plan to do some testing but it will take some time because I need to prepare and buy some equipment first. I will let you know when I started and share my results here!

**Comment by alf45tar** [comment]
I am working to suppor a new type of pedal: the Analog Pad. it will transform PedalinoMini into a trigger-to-midi device too.

**Comment by Steffinuss** [comment]
https://github.com/alf45tar/PedalinoMini/assets/30431113/ff2c9814-e241-4285-a7fb-f606739d9dc9

Here is a short Video of my first test run. It seems quite promising.

Here are my settings:

<img width="318" alt="Bildschirmfoto 2023-12-12 um 22 24 31" src="https://github.com/alf45tar/PedalinoMini/assets/30431113/75c0f821-fdfe-42c4-a44d-6a75d42cb71f">

<img width="424" alt="Bildschirmfoto 2023-12-12 um 22 24 45" src="https://github.com/alf45tar/PedalinoMini/assets/30431113/7d4bad64-7399-42ae-b08f-a2b0fe47b44f">

(In the video I used Note 38 of course)

Here are some basic Midi Notes used in the Garageband Drumset:

![IMG_38B3691BDCB3-1](https://github.com/alf45tar/PedalinoMini/assets/30431113/e5c9dc21-7a52-4371-8991-50860c8f8ee1)

**Comment by Steffinuss** [comment]
Now I analysed the recording from Garageband for iOS in Logic Pro on macOS, that gives me more detailed information about velocity values. Listen to my screen recording here:


https://github.com/alf45tar/PedalinoMini/assets/30431113/67456be6-945b-479d-8987-297e49e42e09
